src/Components/Common/RoleButton.tsx

```diff
@@ -0,0 +1,84 @@
+import React from "react";
+import { useSelector } from "react-redux";
+import Button from "@material-ui/core/Button";
+import { ButtonProps } from "@material-ui/core/Button";
+
+// role types to manage button disable functionality
+// readOnly = all user types with read only access
+export type roleType = "readOnly";
+
+// render different buttons
+export type buttonType = "html" | "materialUI";
+
+/*
+Component to render button "enable" or "disable" based
+on user role
+*/
+
+// Decide whether to disable button or not
```

Comment:
```suggestion
export type roleType = "readOnly";
export type buttonType = "html" | "materialUI";


```
_comments seems excessive here, so please do remove them_